### ISSUE TITLE
Bookmarks not showing

### ISSUE THREAD

**Issue by alanloi** [tintinweb/vscode-inline-bookmarks#16]
Bookmarks are not appearing on the panel.

Getting this error each time i click on a document in the editor:

```
[2020-04-21 20:28:57.283] [exthost] [error] TypeError: Cannot read property 'parent' of undefined
	at InlineBookmarkTreeDataProvider.getParent (c:\Users\Dev\.vscode\extensions\tintinweb.vscode-inline-bookmarks-0.0.16\src\features\inlineBookmarks.js:401:24)
	at c:\Users\Dev\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:573:427
	at c:\Users\Dev\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:54:761
	at new Promise (<anonymous>)
	at Object.t.asPromise (c:\Users\Dev\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:54:733)
	at e.resolveParent (c:\Users\Dev\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:573:395)
	at e.resolveUnknownParentChain (c:\Users\Dev\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:573:116)
	at c:\Users\Dev\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:572:213
```

**Comment by tintinweb** [comment]
hi @alanloi,

thanks for the bug report and sorry for not getting back to you earlier.

the error you're getting is weird. vscode's treeview should actually never call `getParent` with an element that is undefined 🤔

anyway, I am going to fortify the code so that it returns undef in case the caller already provides undef. that should fix the symptom. however, it'd still be interesting to understand why it's being called that way.

can you reliably reproduce?

**Comment by alanloi** [comment]
hi @tintinweb 

Yes, i can reproduce it each time, let me know once you have a fix and i can retest it for you.

I have the same set up on another machine and it seems to work fine on that. Perhaps it's something local.

**Comment by tintinweb** [comment]
also going to add a command that clears the internal state as kind of a last resort. maybe the data the extension was working on was messed up. 

I am going to provide you with a fixed version for testing in this thread later today. Appreciate your help 🙌

**Comment by alanloi** [comment]
No worries, let me know what to do. I'll be here for another few hours at least.

**Comment by tintinweb** [comment]
@alanloi please try this:

* download [vscode-inline-bookmarks-0 0 17 vsix](https://user-images.githubusercontent.com/2865694/81095624-92e04080-8f05-11ea-844e-2471149a8915.png)
* rename to `.vsix`
* `code --install-extension vscode-inline-bookmarks-0.0.17.vsix`

try to reproduce if the issue still persists. if so, try  `cmd + shift + p -> inlineBookmarks.debug.state.reset` and re-open files in the editor that contain bookmarks. the view should autopopulated whenever a file is opened in the editor.

cheers 🙌

**Comment by alanloi** [comment]
@tintinweb 

I've retested and the error is gone. Thanks for your help.

The other thing i noticed earlier was, i reinstalled v16 and the bookmarks did not work at first. I restarted vscode and still no luck. Then i switched to a different project and it started to recognise the inline bookmark comments.

I haven't been able to reproduce it failing again, but it seems to be all working fine now, even before i installed the v17 update.
